### PR TITLE
Reference applications and devices by name instead of id

### PR DESCRIPTION
**Pull request by jviotti**

### CODE REVIEW

File: lib/models/device.coffee [hunk 1]
```diff
@@ -96,23 +130,46 @@ exports.getAllByApplication = (applicationId, callback) ->
...
+
+	username = token.getUsername()
+
+	if not username?
+		return callback(new errors.ResinNotLoggedIn())
+
 	return pine.get
 		resource: 'device'
-		id: deviceId
+		name: name
```
Comment (Page-) [inline lib/models/device.coffee]: This needs to be in a filter

File: lib/models/device.coffee [hunk 2]
```diff
@@ -131,17 +188,33 @@ exports.get = (deviceId, callback) ->
...
+	if not _.isFunction(callback)
+		throw new errors.ResinInvalidParameter('callback', callback, 'not a function')
+
+	if not token.getUsername()?
+		return callback(new errors.ResinNotLoggedIn())
+
 	return pine.delete
 		resource: 'device'
-		id: id
+		name: name
```
Comment (Page-) [inline lib/models/device.coffee]: This needs to be in a filter

File: lib/models/device.coffee [hunk 3]
```diff
@@ -193,12 +282,34 @@ exports.identify = (uuid, callback) ->
...
+	if not _.isFunction(callback)
+		throw new errors.ResinInvalidParameter('callback', callback, 'not a function')
+
+	if not token.getUsername()?
+		return callback(new errors.ResinNotLoggedIn())
+
 	return pine.patch
 		resource: 'device'
-		id: id
+		name: name
```
Comment (Page-) [inline lib/models/device.coffee]: This needs to be in a filter

File: lib/models/device.coffee [hunk 4]
```diff
@@ -212,19 +323,41 @@ exports.rename = (id, name, callback) ->
...
```
Comment (Page-) [inline lib/models/device.coffee]: This needs to be in a filter

File: lib/models/device.coffee (hunk 1 above)
Comment (jviotti) [inline lib/models/device.coffee]: Fixed.

File: lib/models/device.coffee (hunk 2 above)
Comment (jviotti) [inline lib/models/device.coffee]: Fixed.

File: lib/models/device.coffee (hunk 3 above)
Comment (jviotti) [inline lib/models/device.coffee]: Fixed.

**Comment by jviotti** [comment]
@Page- Can you re check that the filters are applied correctly on the places you commented please?

**Pull request merged**